clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -951,7 +978,13 @@ private Record nextFetchedRecord() {
                         return null;
                     }
                     currentBatch = batches.next();
+
                     maybeEnsureValid(currentBatch);
+
+                    if (isolationLevel == IsolationLevel.READ_COMMITTED && isBatchAborted(currentBatch)) {
+                        continue;
```

Comment:
When we skip a batch, we should ensure that `nextFetchOffset` is still updated. We can write a test case to assert the right offset position after receiving a record set where the abort marker is the last entry.